### ISSUE TITLE
Displaying contents of dask variables does not work

### ISSUE THREAD

**Issue by bsesar** [spyder-ide/spyder#14265]
Dear Spyder developers,

The code given below triggers the following error message in Spyder 4.2

```
 File "C:\Users\xxx\Anaconda3\envs\work\lib\site-packages\spyder\plugins\variableexplorer\widgets\namespacebrowser.py", line 408, in set_data
    if data != self.editor.source_model.get_data():
  File "C:\Users\xxx\Anaconda3\envs\work\lib\site-packages\dask\delayed.py", line 573, in __bool__
    raise TypeError("Truth of Delayed objects is not supported")
TypeError: Truth of Delayed objects is not supported
```

Here is the code that can be used to reproduce the problem:
```
import dask.dataframe as dd
import numpy as np
import pandas as pd

df = pd.DataFrame({'a':np.arange(10), 'b':np.arange(10)})
ddf = dd.from_pandas(df, npartitions=1)

ddf
```

The above error message can be avoided by checking "Exclude unsupported data types" in Preferences/Variable explorer.

**Comment by ccordoba12** [comment]
Hey @bsesar, thanks for reporting. This is a serious bug, we'll try to fix it in our next release.

**Comment by Gordon8919** [comment]
@ccordoba12 I am running into the same issue on Spyder 4.2.1, bumping this thread as the bug is still relevant.

**Comment by ccordoba12** [comment]
This is a duplicate of issue #16844 and it'll be fixed in our next version (5.3.0), to be released at the end of the month.
